sentry-ruby/lib/sentry/transaction.rb

```diff
@@ -257,7 +261,7 @@ def finish(hub: nil, end_timestamp: nil)
         @name = UNLABELD_NAME
       end
 
-      @profiler.stop
+      @hub.stop_profiler!(self)
```

Comment:
Given that the hub is responsible for starting a profiler, making this change actually felt like an improvement. Transactions are coupled to hub anyway.